pkg/experiment/metastore/client/methods.go

```diff
@@ -3,23 +3,25 @@ package metastoreclient
 import (
 	"context"
 	"fmt"
-	compactorv1 "github.com/grafana/pyroscope/api/gen/proto/go/compactor/v1"
-	metastorev1 "github.com/grafana/pyroscope/api/gen/proto/go/metastore/v1"
-	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
+	"math/rand"
+	"time"
+
 	"github.com/hashicorp/raft"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	"math/rand"
-	"time"
+
+	compactorv1 "github.com/grafana/pyroscope/api/gen/proto/go/compactor/v1"
+	metastorev1 "github.com/grafana/pyroscope/api/gen/proto/go/metastore/v1"
+	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
 )
 
 func invoke[R any](ctx context.Context, cl *Client,
 	f func(ctx context.Context, instance instance) (*R, error),
 ) (*R, error) {
 	const (
 		n        = 50
-		backoff  = 11 * time.Millisecond
+		backoff  = 51 * time.Millisecond
 		deadline = 500000000 * time.Millisecond
```

Comment:
Not related to your changes, but I think I forgot to change this value after debugging, it should be smaller, like 5s or 50*51=2550ms

Reply:
We should have an exponent here, but even 50ms is not too harmful. I tested various failure cases and it did work well in most of them

Reply:
I have some distant plans to delegate this retries to gRPC conn implementation – I'm thinking about custom resolver and "picker"